linkerd/cache/src/lib.rs

```diff
@@ -9,6 +10,8 @@ use std::hash::Hash;
 use std::sync::{Arc, Weak};
 use tracing::{debug, trace};
 
+#[cfg(all(feature = "unstable", test))]
+mod benches;
```

Comment:
Instead of a module, let's put the benchmarks in a separate `benches` directory inside the `cache` crate? That's generally considered more conventional, and would let ets us run them with `cargo bench`, as well as removing the necessity of an `unstable` feature flag.